### ISSUE TITLE
Build in Prod Mode failed 

### ISSUE THREAD

**Issue by harryhahn** [dominique-mueller/angular-notifier#7]
When i build in --Prod Mode the build failed with:
`ERROR in Error encountered resolving symbol values statically. Could not resolve type NotifierOptions (position 116:37 in the original .ts file), resolving symbol NotifierConfig in C:/Entwicklung/Projekte/marenta/marenta/node_modules/angular-n
otifier/src/models/notifier-config.model.d.ts
`

ng b --prod  => ERROR
 
When disable AOT it build without ERROR

ng b --prod --aot=false  => OK

Thanks

**Comment by dominique-mueller** [comment]
Could you add some details about how you use the library, in particular whether - and if how - you define custom options for this library? This would really help me trying to reproduce your issues. Thanks!

**Comment by harryhahn** [comment]
in app.modules.ts

```
import { NotifierModule, NotifierOptions } from 'angular-notifier';
...
export const  customNotifierOptions: NotifierOptions =
  {  position: {
    horizontal: {
      position: 'right',
      distance: 20
    },
    vertical: {
      position: 'top',
      distance: 100,
      gap: 10
    }
  },
  theme: 'material',
  behaviour: {
    autoHide: 4000,
    onClick: false,
    onMouseover: 'pauseAutoHide',
    showDismissButton: true,
    stacking: 4
  },
  animations: {
    enabled: true,
    show: {
      preset: 'slide',
      speed: 300,
      easing: 'ease'
    },
    hide: {
      preset: 'fade',
      speed: 300,
      easing: 'ease',
      offset: 50
    },
    shift: {
      speed: 300,
      easing: 'ease'
    },
    overlap: 150
  }
};

...
imports: [
...
    NotifierModule.forRoot(customNotifierOptions),
`
```
i use it without customNotifierOptions. same issues.

in my start.component.ts
```

`import { NotifierService } from 'angular-notifier';

@Component({
  selector: 'app-start',
  templateUrl: './start.component.html',
  styleUrls: ['./start.component.css']
})
export class StartComponent implements OnInit {
  private notifier: NotifierService;
  
  constructor(private af: AngularFire, private authService: AuthService,
               notifier: NotifierService
  ) {
     this.notifier =  notifier;
  }

...
  logoutUser( ){
  ..
    this.notifier.notify( 'success', 'Benutzer erfolgreich abgemeldet.' );
  }
`
```
I hope it help.
Thanks!

here the exact Error Message
```

`ERROR in Error encountered resolving symbol values statically. Could not resolve type NotifierOptions (position 116:37 in the original .ts file), resolving symbol NotifierConfig in C:/Entwicklung/Projekte/marenta/marenta/node_modules/angular-no
tifier/src/models/notifier-config.model.d.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in 'C:\Entwicklung\Projekte\marenta\marenta\src'
 @ ./src/main.ts 5:0-74
 @ multi ./src/main.ts
`
```
```

**Comment by dominique-mueller** [comment]
May be related to https://github.com/angular/angular/issues/12631, which then again was fixed 5 days ago by https://github.com/angular/angular/pull/14894.

Will dive deeper into this issue during the upcoming days.

**Comment by dominique-mueller** [comment]
It actually was the issue referenced above - it got fixed for Angular 4. I've implemented a fix for angular-notifier to also make AoT compilation work with Angular 2, and will release version 1.0.4 within the next few minutes.

**Comment by dominique-mueller** [comment]
Just released version 1.0.4 to NPM - @harryhahn, can you verify that everything is working fine for you now?

**Comment by harryhahn** [comment]
Great
it works fine:
Thanks!
